### PR TITLE
KerasClassifier.score: Support `weighted_metrics`

### PR DESCRIPTION
**Pull request by eliadl**
**Background:**

`tf.keras.wrappers.scikit_learn.KerasClassifier` is a wrapper for `tf.keras.Model`.

**Problem:**

When [`tf.keras.Model.compile`](https://www.tensorflow.org/api_docs/python/tf/keras/Model#compile) is called with `weighted_metrics`,
it automatically [adds a `'weighted_'` prefix](https://github.com/tensorflow/tensorflow/blob/r1.14/tensorflow/python/keras/engine/training_utils.py#L796) to the metric's name.
I.e. `'acc'` becomes `'weighted_acc'`.
Even when the metric's `name` is set manually via its class initializer.

`KerasClassifier.score` on the other hand cares only for the metrics `'accuracy'` or `'acc'`,
but sometimes `tf.keras.Model.compile` is called only with `weighted_metrics` and not `metrics`.
And so these metrics are not found and an error is raised.

**Proposed solution:**

`KerasClassifier.score` should look for `'weighted_accuracy'` and `'weighted_acc'` too.

### CODE REVIEW

**Comment by rthadur** [comment]
Closing this PR as this not against `master`, please open a new PR against `master` 
CC @mihaimaruseac

**Comment by mihaimaruseac** [comment]
@eliadl Please make this PR against master and if you also want the changes in the 1.14 branch you can make a cherry-pick against it. The window for cherry-picks for 1.14.1 patch release is just until Wednesday.
